lib/hanami/view/application_context.rb

```diff
@@ -20,18 +22,52 @@ def included(context_class)
 
       def define_initialize
         inflector = application.inflector
+        settings = application[:settings] if application.key?(:settings)
         routes = application[:routes_helper] if application.key?(:routes_helper)
+        assets = application[:assets] if application.key?(:assets)
 
         define_method :initialize do |**options|
           @inflector = options[:inflector] || inflector
+          @settings = options[:settings] || settings
           @routes = options[:routes] || routes
+          @assets = options[:assets] || assets
           super(**options)
         end
       end
 
       module InstanceMethods
         attr_reader :inflector
         attr_reader :routes
+        attr_reader :settings
+
+        def initialize(**args)
+          defaults = {content: {}}
+
+          super(**defaults.merge(args))
+        end
+
+        def content_for(key, value = nil, &block)
+          content = _options[:content]
+          output = nil
+
+          if block
+            content[key] = yield
+          elsif value
+            content[key] = value
+          else
+            output = content[key]
+          end
+
+          output
+        end
+
+        def current_path
+          request.fullpath
+        end
+
+        def csrf_token
+          request.session[Hanami::Action::CSRFProtection::CSRF_TOKEN]
```

Comment:
The fact that this references out to `Hanami::Action` from within hanami-view is definitely a smell to me, and yet another sign that we should just pull all of these out into the hanami gem itself. Looking forward to doing this soon.

In the meantime, I presume we're happy with this constant being referenced without a corresponding `require` given that this is intended only to be used inside a full app with all gems present, yes?

Reply:
I assume that views are present only when actions are there too. But not viceversa (views without actions).